projeuler22.rb

```diff
@@ -0,0 +1,62 @@
+# Using names.txt (right click and 'Save Link/Target As...'), a 46K text file containing over five-thousand first names, begin by sorting it into alphabetical order.
+# Then working out the alphabetical value for each name, multiply this value by its alphabetical position in the list to obtain a name score.
+# For example, when the list is sorted into alphabetical order, COLIN which is worth 3 + 15 + 12 + 9 + 14 = 53, is the 938th name in the list. So, COLIN would obtain a score of 938 × 53 = 49714.
+# What is the total of all the name scores in the file?
+require "pry"
+
+alphabet_arr = ("A".."Z").to_a
+n = 0
+HASH_CONVERSION = {}
+alphabet_arr.each do |l|
+  n += 1
+  HASH_CONVERSION[l] = n
+end
+
+module CONVERT_CHARA
+  def convert_name(name)
+    char_arr = name.split(//).to_a
+    @num_value = []
+    char_arr.each do |char|
+      @num_value << HASH_CONVERSION[char]
+    end
+    @num_sum = @num_value.sum
+  end
+end
+
+class NameCalculator
+  attr_reader :namelist
+  def initialize(namelist)
+    @namelist = File.read(namelist).scan(/\w+/).sort
+  end
+
+  def name_scores
+    extend CONVERT_CHARA
+    @alpha_value = 0
```

Comment:
You seem to use instance variables pretty much everywhere, when you don't need to.

The following isn't 100% accurate, but it's mostly what you need to know for now.

Local `variables` remain in scope inside a method body

Instance `@variables` remain in scope in an instance of a class, and so should be used when a variable is to be used throughout the class

Don't worry about class `@@variables`, they're _very_ rarely used